adapters/repos/db/vector/spfresh/search.go

```diff
@@ -83,7 +83,7 @@ func (s *SPFresh) SearchByVector(ctx context.Context, vector []float32, k int, a
 			}
 
 			// skip duplicates
-			if visited.Visited(id) {
+			if visited.Visited(id) || s.VersionMap.Get(id) != v.Version() {
```

Comment:
We should not skip this vector, the vector itself is good and can be used for the search. The fact that it's not in its ideal posting doesn't really matter for search. All we care about is to check if it's already deleted.